diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1234567..abcdefg 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -234,6 +234,8 @@ ExternalProject_Add(triton-server
   BINARY_DIR "${CMAKE_CURRENT_BINARY_DIR}/triton-server"
   CMAKE_CACHE_ARGS
     -DProtobuf_DIR:PATH=${_FINDPACKAGE_PROTOBUF_CONFIG_DIR}
+    -DProtobuf_INCLUDE_DIR:PATH=${TRITON_THIRD_PARTY_INSTALL_PREFIX}/protobuf/include
+    -DProtobuf_PROTOC_EXECUTABLE:FILEPATH=${TRITON_THIRD_PARTY_INSTALL_PREFIX}/protobuf/bin/protoc-3.21.12.0
     ${_CMAKE_ARGS_OPENSSL_ROOT_DIR}
     ${_CMAKE_ARGS_CMAKE_TOOLCHAIN_FILE}
     ${_CMAKE_ARGS_VCPKG_TARGET_TRIPLET}